pkg/client/kubeconfig_test.go

```diff
@@ -0,0 +1,270 @@
+						KeyData:  keyPem,
+					},
+				}
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(true))
+				Expect(err).To(BeNil())
+			})
+		})
+
+		When("kubeconfg is empty", func() {
+			It("should return false and have error", func() {
+				_, err := os.Create(validKubeConfigPath) // overwrite the kubeconfig file with an empty file
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(ContainSubstring("unable to load kubeconfig"))
+			})
+		})
+
+		When("kubeconfigpath is malformed", func() {
+			It("should return false and not error", func() {
+				longPath := strings.Repeat("a", 1<<16) // a string with 65536 characters
+				isValid, err := isKubeConfigStillValid(longPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).To(BeNil())
+			})
+		})
+
+		When("keyPem does not belong to certPem", func() {
+			It("should return false and have error", func() {
+				certPem, _, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(200, 0, 0))
+				Expect(err).To(BeNil())
+				_, differentKeyPem, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(200, 0, 0))
+				Expect(err).To(BeNil())
+
+				bootstrapClientConfig := &restclient.Config{
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  differentKeyPem,
+					},
+				}
+
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).ToNot(BeNil())
+				Expect(err.Error()).To(ContainSubstring("private key does not match public key"))
+			})
+		})
+
+		When("certPem is expired", func() {
+			It("should return false and not have an error", func() {
+				certPem, keyPem, err := generateMockCertPEMWithExpiration("test", "test", time.Now().AddDate(0, 0, -1))
+				Expect(err).To(BeNil())
+
+				bootstrapClientConfig := &restclient.Config{
+					Host: "https://your-k8s-cluster.com",
+					TLSClientConfig: restclient.TLSClientConfig{
+						CertData: certPem,
+						KeyData:  keyPem,
+					},
+				}
+				err = writeKubeconfigFromBootstrapping(bootstrapClientConfig, validKubeConfigPath, false)
+				Expect(err).To(BeNil())
+
+				isValid, err := isKubeConfigStillValid(validKubeConfigPath, tlsBootstrapClient.logger)
+				Expect(isValid).To(Equal(false))
+				Expect(err).To(BeNil())
+			})
+		})
+
+		When("Some information in bootstrapClientConfig is incorrect", func() {
```

Comment:
test name here is a bit ambiguous - we also won't have kubeconfigs that specify an exec plugin so probably not worth testing this particular failure mode of bad exec plugin api version

Reply:
I can fix the name, but I made this test more for coverage as it specifically hits a error block that nothing else does. So I could remove, but coveralls wouldn't be happy lol

Reply:
fair enough, stuff like this is why I'd like to somehow transition to acceptance tests at some point